pkg/common/catalog/closers.go

```diff
@@ -31,3 +33,23 @@ func (fn closerFunc) Close() error {
 	fn()
 	return nil
 }
+
+func gracefulStopWithTimeout(s *grpc.Server) bool {
+	done := make(chan struct{})
+
+	go func() {
+		s.GracefulStop()
+		close(done)
+	}()
+
+	t := time.NewTimer(time.Minute)
```

Comment:
One problem with this fixed timeout is that the plugin may have defined a larger timeout for an RPC, which will not be honored. I'm not sure though if there is an easy solution for this.

Reply:
I guess the way I see it, is that at this point, something is trying to shut the server down. I don't think we should allow the plugin operations to stall that process for very long. One minute already makes me nervous. Also, for most plugin RPCs, a latency of one minute would have adverse effects on normal SPIRE operation, so I'm not sure how much longer we should accommodate.

We do have timeouts defined for all KeyManager operations. We may eventually put timeouts on the rest of the plugin RPCs. Once we do, maybe this timer value can be revisited (or removed)?

Reply:
> at this point, something is trying to shut the server down.

That's fair and makes sense. I didn't considered this from that perspective.
My comment was just to point out the consideration that this timeout will have precedence over the plugin's defined timeout, which now I think that makes sense since there has been an action taken to shut down the server.